nixos/modules/virtualisation/docker-containers.nix

```diff
@@ -0,0 +1,84 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.docker-containers;
+
+  dockerContainer =
+    { name, config, ... }: {
+
+      options = {
+        image = mkOption {
+          type = types.str;
+          description = "Docker image to run.";
+        };
+        args = mkOption {
```

Comment:
Would be better to call this `cmd` for follow the docker nomenclature.

Reply:
aha right! I was mentally following the Kubernetes container API spec, having forgotten Docker's :-P